go/logic/applier.go

```diff
@@ -482,6 +482,7 @@ func (this *Applier) ApplyIterationInsertQuery() (chunkSize int64, rowsAffected
 		if err != nil {
 			return nil, err
 		}
+		defer tx.Rollback()
```

Comment:
@shlomi-noach just make sure "A transaction must end with a call to Commit or Rollback."